scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1415,32 +1415,47 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) {
   }
 
   private def condExprWithBody[T <: Token: ClassTag]: (Term, Term) = {
-    val cond =
+    val (cond, bodyOpt) =
       if (!dialect.allowQuietSyntax)
-        try condExpr()
+        try (condExpr(), None)
         finally newLinesOpt()
       else if (!token.is[LeftParen])
-        try expr()
+        try (expr(), None)
```

Comment:
```suggestion
        try{ (expr(), None) }
```
looks like Scala 2.12 doesn't like it

Reply:
done, in #3805.